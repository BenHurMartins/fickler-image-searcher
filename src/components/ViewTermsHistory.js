import React, {useEffect} from 'react';
import {View, StyleSheet} from 'react-native';
import {Mixins} from '../styles';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {CardTermHistory} from '../components';
import {useSelector, useDispatch} from 'react-redux';
import * as types from '../reducers/actionsTypes';

const ViewTermsHistory = ({setSearchTerm}) => {
  const dispatch = useDispatch();
  const {termsHistory, searchTerm} = useSelector(state => state.SearchReducer);

  useEffect(() => {
    //Rise the history of terms searched
    getTermsHistory();
  }, []);

  const getTermsHistory = async () => {
    //This section get the stored terms in the users phone
    try {
      let terms = await AsyncStorage.getItem('terms-history');
      terms = terms != null ? JSON.parse(terms) : [];
      dispatch({type: types.SET_TERMS, payload: terms});
    } catch (e) {
      console.log(e);
    }
  };

  const getHintterms = () => {
    //Filter all the terms with the term already used as a search
    let result = termsHistory.filter(str => str.includes(searchTerm)).reverse();
    //Cut the array to show only the 5 newest terms
    result.length = 5;
    return result;
  };
  return (
    <View style={styles.container}>
      {getHintterms().map(element => (
        <CardTermHistory term={element} onPress={setSearchTerm} />
      ))}
    </View>
  );
};

export default ViewTermsHistory;

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    width: Mixins.DEVICE_WIDTH,
    top: 70,
    alignItems: 'center',
    zIndex: 6,
  },
});
