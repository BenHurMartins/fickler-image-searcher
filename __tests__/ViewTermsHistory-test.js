import 'react-native';
import React from 'react';
import {render, act, fireEvent} from '@testing-library/react-native';
import ViewTermsHistory from '../src/components/ViewTermsHistory';
import renderer from 'react-test-renderer';

const setSearchTerm = jest.fn();

describe('ViewTermsHistory component', () => {
  it('ViewTermsHistory renders correctly', () => {
    renderer.create(<ViewTermsHistory setSearchTerm={setSearchTerm} />);
  });
  it('ViewTermsHistory snapshot test', () => {
    const tree = renderer
      .create(<ViewTermsHistory setSearchTerm={setSearchTerm} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Should show history terms', () => {
    const {getByText, getByTestId} = render(
      <ViewTermsHistory setSearchTerm={setSearchTerm} />,
    );
    //the test history terms is already in cache by the reducers
    act(() => {
      expect(getByText('test')).not.toBe(null);
    });
  });
});
