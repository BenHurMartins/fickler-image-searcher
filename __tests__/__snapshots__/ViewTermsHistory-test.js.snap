// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewTermsHistory component ViewTermsHistory snapshot test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "position": "absolute",
      "top": 70,
      "width": 750,
      "zIndex": 6,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#F0F0F0",
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "paddingLeft": 10,
        "width": "90%",
        "zIndex": 9999,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 30,
        }
      }
    >
      test
    </Text>
  </View>
</View>
`;
